plaso/parsers/sqlite_plugins/mac_notes.py

```diff
@@ -0,0 +1,137 @@
+          'ZCONTENTID VARCHAR, ZFILEURL BLOB )'),
+      'ZFOLDER': (
+          'CREATE TABLE ZFOLDER ( Z_PK INTEGER PRIMARY KEY, Z_ENT INTEGER,'
+          'Z_OPT INTEGER, ZACCOUNT INTEGER, Z1_ACCOUNT INTEGER, ZPARENT'
+          'INTEGER, Z6_PARENT INTEGER, ZISDISTINGUISHED INTEGER,'
+          'ZALLEGEDHIGHESTMODIFICATIONSEQUENCE INTEGER,'
+          'ZCOMPUTEDHIGHESTMODIFICATIONSEQUENCE INTEGER, ZUIDNEXT INTEGER,'
+          'ZUIDVALIDITY INTEGER, ZTRASHACCOUNT INTEGER, Z1_TRASHACCOUNT'
+          'INTEGER, ZNAME VARCHAR, ZCHANGEKEY VARCHAR, ZFOLDERID VARCHAR,'
+          'ZSYNCSTATE VARCHAR, ZSERVERNAME VARCHAR )'),
+      'ZNOTE': (
+          'CREATE TABLE ZNOTE ( Z_PK INTEGER PRIMARY KEY, Z_ENT INTEGER,'
+          'Z_OPT INTEGER, ZBODY INTEGER, ZFOLDER INTEGER, Z6_FOLDER'
+          'INTEGER, ZMIMEDATASIZE INTEGER, ZDATECREATED TIMESTAMP,'
+          'ZDATEEDITED TIMESTAMP, ZREMOTEID VARCHAR, ZTITLE VARCHAR,'
+          'ZCHANGEKEY VARCHAR, ZUNIVERSALLYUNIQUEID BLOB )'),
+      'ZNOTEBODY': (
+          'CREATE TABLE ZNOTEBODY ( Z_PK INTEGER PRIMARY KEY, Z_ENT'
+          'INTEGER, Z_OPT INTEGER, ZNOTE INTEGER, Z10_NOTE INTEGER,'
+          'ZHTMLSTRING VARCHAR )'),
+      'ZOFFLINEACTION': (
+          'CREATE TABLE ZOFFLINEACTION ( Z_PK INTEGER PRIMARY KEY, Z_ENT'
+          'INTEGER, Z_OPT INTEGER, ZSEQUENCENUMBER INTEGER, ZACCOUNT'
+          'INTEGER, Z1_ACCOUNT INTEGER, ZFOLDER INTEGER, Z6_FOLDER INTEGER,'
+          'ZPARENT INTEGER, Z6_PARENT INTEGER, ZORIGINALPARENT INTEGER,'
+          'Z6_ORIGINALPARENT INTEGER, ZFOLDER1 INTEGER, Z6_FOLDER1 INTEGER,'
+          'ZNOTE INTEGER, Z10_NOTE INTEGER, ZORIGINALFOLDER INTEGER,'
+          'Z6_ORIGINALFOLDER INTEGER )'),
+      'Z_METADATA': (
+          'CREATE TABLE Z_METADATA (Z_VERSION INTEGER PRIMARY KEY, Z_UUID'
+          'VARCHAR(255), Z_PLIST BLOB)'),
+      'Z_MODELCACHE': ('CREATE TABLE Z_MODELCACHE (Z_CONTENT BLOB)'),
+      'Z_PRIMARYKEY': (
+          'CREATE TABLE Z_PRIMARYKEY (Z_ENT INTEGER PRIMARY KEY, Z_NAME'
+          'VARCHAR, Z_SUPER INTEGER, Z_MAX INTEGER)')
+  }]
+
+  def ParseZHTMLSTRINGRow(self, parser_mediator, query, row, **unused_kwargs):
+    """Parses a row from the database.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      query (str): query that created the row.
+      row (sqlite3.Row): row resulting from query.
+    """
+    # Note that pysqlite does not accept a Unicode string in row['string'] and
+    # will raise "IndexError: Index must be int or string".
+    query_hash = hash(query)
+    event_data = MacNotesZhtmlstringEventData()
+    soup = BeautifulSoup(self._GetRowValue(query_hash, row, 'zhtmlstring'),
+                         'html.parser')
+    body = soup.body.prettify()
+    body = re.sub(r'(<\/?(div|body|span|b|table|tr|td|tbody|p).*>\n?)', '',
+                  body)
+    event_data.zhtmlstring = body
```

Comment:
Rather than doing this, store the zhtmlstring unmodified, then convert it in the formatter.

Reply:
Hey @alimez - it looks like you didn't respond to this comment, what happened?

Reply:
I missed this one. Work in progress.

Reply:
Done.